openedx/core/djangoapps/credit/api.py

```diff
@@ -183,14 +183,13 @@ def create_credit_request(course_key, provider_id, username):
 
     """
     try:
-        user_eligibility = CreditEligibility.objects.select_related('course', 'provider').get(
+        user_eligibility = CreditEligibility.objects.select_related('course').get(
             username=username,
-            course__course_key=course_key,
-            provider__provider_id=provider_id
+            course__course_key=course_key
         )
         credit_course = user_eligibility.course
-        credit_provider = user_eligibility.provider
-    except CreditEligibility.DoesNotExist:
+        credit_provider = credit_course.providers.get(provider_id=provider_id)
+    except CreditEligibility.DoesNotExist, CreditProvider.DoesNotExist:
```

Comment:
Discussed offline: one of the changes in this PR is to make `CreditEligibility` eligibility for all providers in a course rather than for a specific provider.  Deleting the FK from `CreditEligibility` to `CreditProvider` makes this change necessary.